velox/core/PlanNode.cpp

```diff
@@ -332,9 +332,6 @@ AbstractJoinNode::AbstractJoinNode(
       leftKeys_.size(),
       rightKeys_.size(),
       "JoinNode requires same number of join keys on left and right sides");
-  if (isSemiJoin() || isAntiJoin()) {
-    VELOX_CHECK_NULL(filter, "Semi and anti join does not support filter");
```

Comment:
all joins seem to support a filter.
This is dead code, to begin with since the member `filter_` has the filters.